serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/ObjectInspectorUtils.java

```diff
@@ -1263,6 +1267,33 @@ public static int compare(Object o1, ObjectInspector oi1, Object o2,
     }
   }
 
+  public static boolean isConversionSupported(ObjectInspector oi1, ObjectInspector oi2) {
+    if (oi1.getCategory() == oi2.getCategory()) {
+      if (oi1.getCategory() == Category.PRIMITIVE) {
+        PrimitiveObjectInspector poi1 = ((PrimitiveObjectInspector) oi1);
+        PrimitiveObjectInspector poi2 = ((PrimitiveObjectInspector) oi2);
+        if ((poi1.getPrimitiveCategory().equals(PrimitiveObjectInspector.PrimitiveCategory.VARCHAR)
+            && poi2.getPrimitiveCategory().equals(PrimitiveObjectInspector.PrimitiveCategory.STRING)) || (
+            poi1.getPrimitiveCategory().equals(PrimitiveObjectInspector.PrimitiveCategory.STRING)
+                && poi2.getPrimitiveCategory().equals(PrimitiveObjectInspector.PrimitiveCategory.VARCHAR))) {
+          return true;
```

Comment:
What's the reason why we need this? I feel this is a little risky since it affect several places.

Reply:
Varchar and string should be convertible and be able to compare. Moving this to base class of Array UDFs, AbstractGenericUDFArrayBase